src/git/repo.rs

```diff
@@ -1446,6 +1711,152 @@ impl<'repo> Branch<'repo> {
     }
 }
 
+/// A Git file status indicator.
+/// See https://git-scm.com/docs/git-status#_short_format
+#[allow(missing_docs)]
+#[derive(Debug, PartialEq, Eq)]
+pub enum FileStatus {
+    Unmodified,
+    Modified,
+    Added,
+    Deleted,
+    Renamed,
+    Copied,
+    Unmerged,
+    Untracked,
+    Ignored,
+}
+
+/// The status of a file in the repo.
+#[derive(Debug, PartialEq, Eq)]
+pub struct StatusEntry {
+    /// The status of the file in the index.
+    pub index_status: FileStatus,
+    /// The status of the file in the working copy.
+    pub working_copy_status: FileStatus,
+    /// The file mode of the file in the working copy.
+    pub working_copy_file_mode: FileMode,
+    /// The file path.
+    pub path: PathBuf,
+    /// The original path of the file (for renamed files).
+    pub orig_path: Option<PathBuf>,
+}
+
+impl StatusEntry {
+    /// Returns the paths associated with the status entry.
+    pub fn paths(&self) -> Vec<PathBuf> {
+        let mut result = vec![self.path.clone()];
+        if let Some(orig_path) = &self.orig_path {
+            result.push(orig_path.clone());
+        }
+        result
+    }
+}
+
+fn parse_file_status(status: u8) -> eyre::Result<FileStatus> {
```

Comment:
It would be best to implement `From`/`TryFrom` here.

Reply:
Done